### PR TITLE
fix: List component cursor pointer corrections, move css to styles from Ngx

### PR DESCRIPTION
**Pull request by DeepakSap14**
## Related Issue
Related to Issue https://github.com/SAP/fundamental-ngx/issues/5025

## Description
Moving list css present in Ngx to Styles.
It is moving core list css and platform list css.

#### Please check whether the PR fulfills the following requirements

1. The output matches the design specs
- [x] All values are in `rem`
- [x] Text elements follow the truncation rules
- [x] hover state of the element follow design spec
- [x] focus state of the element follow design spec
- [x] active state of the element follow design spec
- [x] selected state of the element follow design spec
- [x] selected hover state of the element follow design spec
- [x] pressed state of the element follow design spec
- [x] Responsiveness rules - the component has modifier classes for all breakpoints
- [x] Includes Compact/Cosy/Tablet design
- [x] RTL support
2. The code follows fundamental-styles code standards and style
- [x] only one top level `fd-*` class is used in the file
- [x] BEM naming convention is used
- [x] Mixins are used for repeatable code (`fd-rtl`, `fd-ellipsis`, `fd-flex`, `fd-selected`, `fd-focus`, ect.)
- [x] A11y support - keyboard support, screenreader support, proper ARIA attributes, etc.
- [x] `fd-reset()` mixin is applied to all elements
- [x] Variables are used, if some value is used more than twice.
- [x] Checked if current components can be reused, instead of having new code.
3. Testing
- [x] tested Storybook examples with "CSS Resources" `normalize` option 
- [x] tested Storybook examples with "CSS Resources" `unnormalize` option 
- [x] Verified all styles in IE11
- [x] Updated tests
- [x] last commit message should have `[ci visual]` so it can trigger chromatic visual regression (e.g. `test: run chromatic visual regression [ci visual]`)
4. Documentation
- [x] Storybook documentation has been created/updated
- na Breaking Changes [wiki](https://github.com/SAP/fundamental-styles/wiki/Breaking-Changes) has been updated in case of breaking changes.

### CODE REVIEW

File: src/_listByline.scss [hunk 1]
```diff
@@ -101,7 +102,7 @@
       color: $fd-list-byline-section-text-color;
       padding-right: $fd-list-byline-elements-spacing;
       line-height: $fd-list-byline-text-line-height;
-      align-self: flex-end;
+      align-self: flex-start;
```
Comment (InnaAtanasova) [inline src/_listByline.scss:105]: why was this changed?

File: src/_listBase.scss [hunk 2]
```diff
@@ -123,6 +137,14 @@
...
       flex-basis: auto;
     }
+
+    &--growing {
+      @include fd-flex-center();
+    }
+
+    &--unread {
+      font-weight: bold;
+    }
```
Comment (InnaAtanasova) [inline src/_listBase.scss:147]: where are these in the html code? I looked for `__title--growing` and `__title--unread` but couldn't find it. If it's present in NGX but not in styles, you should add examples in fund-styles so we can test it.

File: src/_listBase.scss [hunk 3]
```diff
@@ -100,6 +100,19 @@
...
+    &--inactive {
+      pointer-events: none;
+
+      .#{$block}__title,
+      .#{$block}__secondary {
+        pointer-events: none;
+      }
+    }
+
+    &--no-seprator {
```
Comment (InnaAtanasova) [inline src/_listBase.scss:112]: couldn't find this anywhere in the code as an example

**Review by InnaAtanasova — CHANGES_REQUESTED** [review]
The hover state for selected list item is not working. The background should change:
Screen Shot 2021-05-12 at 2 20 39 PM

File: src/_listBase.scss (hunk 3 above)
Comment (DeepakSap14) [reply src/_listBase.scss:112]: added examples for all css changes.

File: src/_listBase.scss (hunk 2 above)
Comment (DeepakSap14) [reply src/_listBase.scss:147]: added examples.

File: src/_listByline.scss (hunk 1 above)
Comment (DeepakSap14) [reply src/_listByline.scss:105]: i think some of the css present in Ngx is un-necessary and could have been avoided. i am not making any change for that because it is already working and approved code.
This change is present in https://github.com/SAP/fundamental-ngx/blob/main/libs/core/src/lib/list/list.component.scss, so i have moved it to styles.

**Comment by DeepakSap14** [comment]
> The hover state for selected list item is not working. The background should change:
> Screen Shot 2021-05-12 at 2 20 39 PM

This one i am looking, will take little time to identify the fix.

**Comment by DeepakSap14** [comment]
> > The hover state for selected list item is not working. The background should change:
> > Screen Shot 2021-05-12 at 2 20 39 PM
> 
> This one i am looking, will take little time to identify the fix.

@InnaAtanasova ,hover state on selected list item is as well fixed now. please review.

**Comment by InnaAtanasova** [comment]
What is the difference between these?
Screen Shot 2021-05-17 at 11 22 09 AM

Screen Shot 2021-05-17 at 11 22 02 AM

**Comment by InnaAtanasova** [comment]
We already have growing list with busy indicator. Why did you add this one?

Screen Shot 2021-05-17 at 11 22 39 AM
Screen Shot 2021-05-17 at 11 22 31 AM

**Comment by InnaAtanasova** [comment]
We have counter. Why did you add a new one?

Screen Shot 2021-05-17 at 11 22 53 AM
Screen Shot 2021-05-17 at 11 22 46 AM

File: src/_listBase.scss [hunk 4]
```diff
@@ -302,4 +324,21 @@
...
+    font-size: 0.875rem;
+    color: var(--sapContent_MarkerTextColor);
+    padding-left: $fd-list-item-padding-x;
+    text-shadow: none;
+    flex: none;
+
+    @include fd-active() {
+      color: var(--sapHighlightTextColor);
+    }
+  }
```
Comment (InnaAtanasova) [inline src/_listBase.scss:338]: I think we already have a counter.

File: src/_listBase.scss [hunk 5]
```diff
@@ -302,4 +324,21 @@
...
+
+  &__infinite-scroll {
+    overflow: scroll;
+    outline: none;
+  }
```
Comment (InnaAtanasova) [inline src/_listBase.scss:343]: I couldn't find an example for this

**Comment by DeepakSap14** [comment]
Need to refactor platform list component in Ngx, so already existing classes can be used.
once that is completed, remaining css classes will be moved here.

File: src/_listBase.scss (hunk 5 above)
Comment (DeepakSap14) [reply src/_listBase.scss:319]: added example

File: src/_listByline.scss [hunk 6]
```diff
@@ -37,7 +37,7 @@
       border-radius: $fd-list-byline-thumbnail-border-radius;
       margin-right: $fd-list-byline-thumbnail-spacing;
       font-size: 2.5rem;
-      align-self: flex-start;
+      align-self: center;
```
Comment (DeepakSap14) [inline src/_listByline.scss:40]: here icon/avatar should be centre align, https://wiki.wdf.sap.corp/wiki/display/visualcore/List+%28Fiori+3%29+-+Standard+List+with+Byline.
Borderless byline has issue with flex-start.

File: src/_listByline.scss (hunk 1 above)
Comment (DeepakSap14) [reply src/_listByline.scss]: reverted back now. i do not see any issue using 'flex-end', examples are working fine. so no need to bring Ngx change for this.

File: src/_listBase.scss (hunk 2 above)
Comment (DeepakSap14) [reply src/_listBase.scss]: removed these duplicate classes

File: src/_listBase.scss (hunk 4 above)
Comment (DeepakSap14) [reply src/_listBase.scss]: i have removed all the duplicate classes which was coming from platform list. i have verified Ngx as well using correct styles classes.

File: src/_listByline.scss (hunk 6 above)
Comment (InnaAtanasova) [inline src/_listByline.scss:40]: I don't see this rule centering the icon horizontally. You still have the same:
Screen Shot 2021-05-20 at 1 25 43 PM

You can use `text-align: center`

File: src/_listDefinitions.scss [hunk 7]
```diff
@@ -93,6 +94,7 @@ $semantic-color: (
       left: 0.125rem;
       right: 0.125rem;
       bottom: 0.125rem;
+      pointer-events: none;
```
Comment (InnaAtanasova) [inline src/_listDefinitions.scss:97]: Can you explain what you are achieving with enabling/disabling pointer events here?

**Review by InnaAtanasova — CHANGES_REQUESTED** [review]

File: src/_listByline.scss (hunk 6 above)
Comment (DeepakSap14) [reply src/_listByline.scss:40]: you can observe in this example. (styles docs)
Screenshot 2021-05-20 at 11 04 14 PM

Comment (DeepakSap14) [reply src/_listByline.scss:40]: here Avatar is not centre aligned vertically.

File: src/_listDefinitions.scss (hunk 7 above)
Comment (DeepakSap14) [reply src/_listDefinitions.scss:97]: this fix was done by markowski in core Ngx at https://github.com/SAP/fundamental-ngx/blob/main/libs/core/src/lib/list/list.component.scss#L4. i am not very sure of the reason. i have moved it here from Ngx.

Comment (DeepakSap14) [reply src/_listDefinitions.scss:97]: when navigation list item is in focus, cursor is not changing to "grabbing cursor". by making pointer-events: none, it has "grabbing" cursor.
list-cursor-pointer

**Review by InnaAtanasova — CHANGES_REQUESTED** [review]
Screen Shot 2021-05-25 at 1 31 27 PM

The icon inside the thumbnail is still not centered horizontally

File: src/_listByline.scss (hunk 6 above)
Comment (InnaAtanasova) [reply src/_listByline.scss:40]: You can remove this rule, you don't need to change it to center.

**Comment by DeepakSap14** [comment]
> Screen Shot 2021-05-25 at 1 31 27 PM
> 
> The icon inside the thumbnail is still not centered horizontally

This has been removed. No change here.

**Comment by DeepakSap14** [comment]
> > Screen Shot 2021-05-25 at 1 31 27 PM
> > The icon inside the thumbnail is still not centered horizontally
> 
> This has been removed. keeping flex-start.

**Review by InnaAtanasova — CHANGES_REQUESTED** [review]
When you hover over the text you get the `text` cursor. It should be the default.

**Comment by InnaAtanasova** [comment]
Screen Shot 2021-06-04 at 11 22 53 AM

The thumbnail is still not centered.

**Comment by DeepakSap14** [comment]
> When you hover over the text you get the `text` cursor. It should be the default.

If you implementation of Core list in fundamental-ngx, `text` cursor is used.
Do you want me to revert that and use default cursor?

**Comment by DeepakSap14** [comment]
> Screen Shot 2021-06-04 at 11 22 53 AM
> 
> The thumbnail is still not centered.

you did asked to remove it here https://github.com/SAP/fundamental-styles/pull/2343#discussion_r639021611.

Now i am placed it in vertical and horizontal center, fiori also says it. Hope this is correct.

**Comment by InnaAtanasova** [comment]
> #2343 (comment)

Screen Shot 2021-06-07 at 9 43 11 AM

I did not ask such thing! I said that this rule that you applied is NOT centering the element HORIZONTALLY and even suggested how to fix this using `text-align: center`.

**Review by InnaAtanasova — APPROVED** [review]

**Comment by DeepakSap14** [comment]
> > #2343 (comment)
> 
> Screen Shot 2021-06-07 at 9 43 11 AM
> 
> I did not ask such thing! I said that this rule that you applied is NOT centering the element HORIZONTALLY and even suggested how to fix this using `text-align: center`.

I think, I misunderstood and removed vertical centre as well. i think, you wanted to say to remove `align-self: center;` property and use mixin, which will place horizontal and vertical centre both.
